### PR TITLE
Ansible vagrant init

### LINKED ISSUE #8: Add Vagrant and Ansible
The project needs a standard Vagrantfile that uses a shell provisioner that calls out to the script/setup function, which will in turn use Ansible to bootstrap and then setup the box.

The bootstrap script should also be able to call out to Ansible to make sure that the dependencies are brought up to date.

### PR DESCRIPTION
**Pull request by oakensoul**
## Overview
I have added Vagrant and Ansible to the project.

Resolves #8

### Demo

Optional. Screenshots, `curl` examples, etc.

### Notes
The Vagrant file calls out to the script/bootstrap file for provisioning.

The bootstrap script then uses apt to update and make sure all Ansible requirements exist on the box. This is hopefully a temporary solution. I would like to create a Python Development box so that all the ansible/infrastructure stuff related to generic “python development” is covered by Packer generating a box.

From there, the bootstrap script calls out to the requirements.yml for any Ansible Galaxy roles we might need. It then calls the environment specific playbook (in this case vagrant.yml).

Once all that is done, we’ve got the basics of a Python dev environment built up with AWS, BOTO, etc.

## Testing Instructions

 * Make sure you have the project set up according to the README
 * Run `vagrant status` to make sure you don’t have an environment already running. If you do, `vagrant destroy` it.
 * Run `vagrant up`, the script should execute properly
 * Run `vagrant ssh`
 * Run “aws help” and you should receive the appropriate prompts from the AWS command line, meaning the install worked correctly.

### CODE REVIEW

File: ansible/roles/base/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+
+---
+
+    - name: Update apt cache
+      apt: update-cache=yes cache_valid_time=86400
+      become: yes
+      become_method: sudo
```
Comment (asagage) [inline ansible/roles/base/tasks/main.yml]: You don't need the become: yes and become_method here as you are setting them in vagrant.yml

File: script/bootstrap [hunk 2]
```diff
@@ -4,4 +4,78 @@
...
+# we're using and not into the project itself.
+# <TemporaryProvisioning>
+echo "Provsioning the Server:"
+export DEBIAN_FRONTEND=noninteractive
+
+echo "  1/5. Update apt"
+apt-get -qq update
+
+echo "  2/5. Upgrading all packages"
+apt-get -qq -y dist-upgrade
```
Comment (asagage) [inline script/bootstrap]: I think the -y is implied by the -qq

File: script/bootstrap [hunk 3]
```diff
@@ -4,4 +4,78 @@
...
+
+if [ $(dpkg-query -W -f='${Status}' ansible 2>/dev/null | grep -c "ok installed") -eq 0 ];
+then
+    echo "  3/5. Add Ansible PPA"
+    apt-add-repository ppa:ansible/ansible
+
+    echo "  4/5. Install python2 and ansible"
+    apt-get -qq -y install python python-dev python-pip ansible ntp
```
Comment (asagage) [inline script/bootstrap]: -qq = (-qq -y)

File: script/bootstrap [hunk 4]
```diff
@@ -4,4 +4,78 @@
...
+else
+    echo "  3/5. Add Ansible PPA (Skipped, already installed)"
+    echo "  4/5. Install python2 and ansible (Skipped, already installed)"
+fi
+
+echo "  5/5. Remove auto-installed packages that are no longer required"
+apt-get -qq -y autoremove
```
Comment (asagage) [inline script/bootstrap]: why -y? see -qq

**Review by asagage — COMMENTED** [review]
i'd cleanup the playbook a bit and test without the -y's

File: ansible/roles/base/tasks/main.yml (hunk 1 above)
Comment (oakensoul) [reply ansible/roles/base/tasks/main.yml]: That is true when using vagrant.yml, but if there were another playbook using this role other than vagrant, you can’t guarantee that… I’m not sure which approach would actually be more appropriate there… which is part of the reason I left it in both. It’s probably more correct to remove it from the vagrant.yml file...

Comment (oakensoul) [reply ansible/roles/base/tasks/main.yml]: Thinking about it, I think what should happen is the become_method is set at the playbook level. Whether to escalate or not should still be attached to the task.

File: README.md [hunk 5]
```diff
@@ -1,7 +1,12 @@
 # pyramid-zappa-api-boilerplate
 A Python boilerplate for creating AWS Lambda API projects using Pyramid and Zappa, orchestrated with Ansible.
 
-Before participating in our communit, please read and agree to our [Code of Conduct](.github/CODE_OF_CONDUCT.md).
+## Policies
+Before participating in our community, please read through and make sure you fundamentally agree with our various
+policies and guides. If you have any questions or concerns, please feel free to let us know.
+ * [Code of Conduct](.github/CODE_OF_CONDUCT.md)
+ * [Contributing Guideliens](.github/CONTRIBUTING.md).
```
Comment (ajbosco) [inline README.md]: Typo Police: Guidelines

File: script/bootstrap [hunk 6]
```diff
@@ -4,4 +4,78 @@
...
+
+if [ $(dpkg-query -W -f='${Status}' ansible 2>/dev/null | grep -c "ok installed") -eq 0 ];
+then
+    echo "  2/5. Add Ansible PPA"
+    apt-add-repository ppa:ansible/ansible
+
+    echo "  3/5. Install python2 and ansible"
+    apt-get -y install python python-dev python-pip ansible ntp
+else
+    echo "  2/5. Add Ansible PPA (Skipped, already installed)"
```
Comment (ajbosco) [inline script/bootstrap:58]: These numbers are out of order.

**Review by ajbosco — DISMISSED** [review]

File: script/bootstrap (hunk 6 above)
Comment (oakensoul) [reply script/bootstrap:58]: It’s an if/then statement :)  Steps 2/3 repeat depending on which part of the if block it falls in. Unless I’m missing something?

Comment (ajbosco) [reply script/bootstrap:58]: It would help if I could read....

**Review by ajbosco — APPROVED** [review]

**Review by asagage — APPROVED** [review]

File: Vagrantfile [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+# configures the configuration version (we support older styles for
+# backwards compatibility). Please don't change it unless you know what
+# you're doing.
+Vagrant.configure(2) do |config|
+  # Get rid of that pesky "stdin: is not a tty" error
+  config.ssh.shell = "bash -c 'BASH_ENV=/etc/profile exec bash'"
+
+  # Every Vagrant development environment requires a box. You can search for
+  # boxes at https://atlas.hashicorp.com/search.
+  config.vm.box = "bento/ubuntu-16.04"
```
Comment (markdia) [inline Vagrantfile:22]: sweet. double checking the version here.  latest LTS release.  check.

**Review by markdia — COMMENTED** [review]
Ubuntu 16.04 comes by default with Python 3.5.1 installed as the python3 binary.  We doing anything to enforce a specific version across folks' environments?  Earlier you had stated your desire for 3.6 for Lambda support.

File: Vagrantfile (hunk 7 above)
Comment (oakensoul) [reply Vagrantfile:22]: It’s Issue #11, already up for the next branch once this one is finished.

Comment (markdia) [reply Vagrantfile:22]: nice!

**Review by markdia — APPROVED** [review]
with issue 11 on the way i think i'm good!

**Pull request merged**